test/Npgsql.Tests/Types/JsonDynamicTests.cs

```diff
@@ -327,22 +341,36 @@ public async Task Poco_unspecified_polymorphic_mapping()
             TemperatureC = 10
         };
 
-        var sql = """{"$type":"extended","TemperatureF":49,"Date":"2019-09-01T00:00:00","TemperatureC":10,"Summary":"Partly cloudy"}""";
+        // Note: we assert a specific string representation, though jsonb doesn't guarantee the property ordering; so the assert may break
+        // for jsonb if PostgreSQL changes its implementation.
+        var sql =
+            IsJsonb
+                ? """{"Date": "2019-09-01T00:00:00", "$type": "extended", "Summary": "Partly cloudy", "TemperatureC": 10, "TemperatureF": 49}"""
```

Comment:
@NinoFloris not sure why we're still seeing `$type` here for jsonb even though AllowOutOfOrderMetadataProperties is disabled...

Reply:
Because it's marked as `JsonDerived`, and it's serializing as itself, the order doesn't matter.

Reply:
I think this is because of the `jsonbClrTypes` argument on the `EnableDynamicJson` method;

If I pass in the type to be serialized, it does not include the `$type` property:

![image](https://github.com/user-attachments/assets/f37e65e5-6361-4821-bbb6-75d94df4d870)